esp32_lyrat/include/bsp/esp32_lyrat.h

```diff
@@ -0,0 +1,352 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: CC0-1.0
+ */
+
+#pragma once
+
+#include "sdkconfig.h"
+#include "driver/gpio.h"
+#include "driver/i2c.h"
+#include "driver/sdmmc_host.h"
+#include "esp_codec_dev.h"
+#include "driver/touch_pad.h"
+#include "iot_button.h"
+
+
+#if ESP_IDF_VERSION < ESP_IDF_VERSION_VAL(5, 0, 0)
+#include "driver/i2s.h"
+#else
+#include "driver/i2s_std.h"
+#endif
+
+/**************************************************************************************************
+ *  Board pinout
+ **************************************************************************************************/
+/* I2C */
+#define BSP_I2C_SCL           (GPIO_NUM_23)
+#define BSP_I2C_SDA           (GPIO_NUM_18)
+
+/* Audio */
+#define BSP_I2S_SCLK          (GPIO_NUM_5)
+#define BSP_I2S_MCLK          (GPIO_NUM_0)
+#define BSP_I2S_LCLK          (GPIO_NUM_25)
+#define BSP_I2S_DOUT          (GPIO_NUM_26)
+#define BSP_I2S_DSIN          (GPIO_NUM_35)  // From ADC ES8388
+#define BSP_POWER_AMP_IO      (GPIO_NUM_21)
+
+/* uSD card */
+#define BSP_SD_D0             (GPIO_NUM_2)
+#define BSP_SD_D1             (GPIO_NUM_4)
+#define BSP_SD_D2             (GPIO_NUM_12)
+#define BSP_SD_D3             (GPIO_NUM_13)
+#define BSP_SD_CMD            (GPIO_NUM_15)
+#define BSP_SD_CLK            (GPIO_NUM_14)
+
+/* Leds */
+typedef enum bsp_led_t {
+    BSP_LED_GREEN = GPIO_NUM_22,
+} bsp_led_t;
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**************************************************************************************************
+ *
+ * Buttons interface
+ *
+ * Example configuration:
+ * \code{.c}
+ * button_handle_t button[BSP_BUTTON_NUM];
+ * for (int i = 0; i < BSP_BUTTON_NUM; i++) {
+ *     button[i] = iot_button_create(&bsp_button_config[i]);
+ * }
+ * \endcode
+ **************************************************************************************************/
```

Comment:
Could we provide all-buttons-init function `esp_err_t bsp_iot_button_create(button_handle_t btn_array[], int *btn_cnt, int btn_array_size);` and update this comment

```
* Example configuration:
 * \code{.c}
 * button_handle_t btns[BSP_BUTTON_NUM];
 * bsp_iot_button_create(btns, NULL, BSP_BUTTON_NUM);
 * iot_button_register_cb(btns[0], ...
 * \endcode
```